### PR TITLE
Certificate of Eligibility | Fix logic issues

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- *(Summarize the changes that have been made to the platform)*
  > - Add "regular restoration of entitlement" option
  > - Rename "restoration of entitlement" to "one-time restoration of entitlement" and move it to be 4th in the list
  > - Make property owned question required
  > - In documented values, set home sell indicator to `null`. The frontend does not pass this value to the backend
  > - Fix loan number duplicates when there are two loans added without an assigned loan number
  > - Code cleanup 
  > - Updated tests
- *(If bug, how to reproduce)*
  > - Go to the [COE form in staging](https://staging.va.gov/housing-assistance/home-loans/request-coe-form-26-1880/)
  > - Log in to user 228
  > - Navigate to question asking if you have a previous loan, choose yes
  > - In the list of properties, fill out required files, but leave the VA loan number empty
  > - Add another property, and leave the VA loan number empty. A loan number is not unique error will appear
- *(What is the solution, why is this the solution)*
  > Validation will now ignore empty VA loan number entries
- *(Which team do you work for, does your team own the maintainence of this component?)*
  > Benefits team 1
- *(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)*
  > Late Jan to early Feb 2023

## Related issue(s)

[#50226](https://github.com/department-of-veterans-affairs/va.gov-team/issues/50226)

## Testing done

- *Describe what the old behavior was prior to the change*
  > Unique VA loan number error when 2 entries are empty
- *Describe the steps required to verify your changes are working as expected*
  > Run unit tests
- *Describe the tests completed and the results*
  > Updated unit test and all passing

### Before

<details><summary>How will you use your COE? (before)</summary>

how will you use your COE? 4 choices shown: A restoration of entitlement, a regular cash-out refinance of a current VA home loan, an interest rate reduction refinance loan to refinance a current VA loan, or an entitlement inquiry only</details>

### After

<details><summary>How will you use your COE? (after)</summary>

How will you use your COE? 5 choices shown: A regular restoration of entitlement, a regular cash-out refinance of a current VA home loan, an interest rate reduction refinance loan to refinance a current VA loan, a one-time restoration of entitlement, or an entitlement inquiry only</details>

<details><summary>Do you still own this property, now required</summary>

Do you still own this property? required, yes or no radios
</details>

## What areas of the site does it impact?

Certificate of Eligibility form (not yet released to production)

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/lgy/coe/form/config/chapters/loans/loanIntent.js [hunk 1]
```diff
@@ -1,24 +1,42 @@
...
-export const schema = loanIntent;
+// replacing loanIntent from schemaImports
+export const schema = {
+  type: 'object',
+  properties: {
+    intent: {
+      type: 'string',
+      enum: [
+        LOAN_INTENT.regular, // new entry, will not be submitted
+        LOAN_INTENT.refiance,
```
Comment (jerekshoe) [inline src/applications/lgy/coe/form/config/chapters/loans/loanIntent.js:13]: *refinance

File: src/applications/lgy/coe/form/validations.js [hunk 2]
```diff
@@ -27,9 +27,9 @@ export const validateDocumentDescription = (errors, fileList) => {
 };
 
 export const validateUniqueVALoanNumber = (errors, data) => {
-  const loans = (data || []).map(loan =>
-    replaceNonDigits(loan?.vaLoanNumber || ''),
-  );
+  const loans = (data || [])
+    .map(loan => replaceNonDigits(loan?.vaLoanNumber || ''))
+    .filter(Boolean);
```
Comment (jerekshoe) [inline src/applications/lgy/coe/form/validations.js:32]: Out of curiosity, what does this part do?

**Review by jerekshoe — APPROVED** [review]

File: src/applications/lgy/coe/form/validations.js (hunk 2 above)
Comment (Mottie) [reply src/applications/lgy/coe/form/validations.js:32]: While testing, I added 2 loans, both with the VA loan number blank and I saw a VA loan number must be unique. So the `filter(Boolean)` is removing empty VA loan numbers from the validation check

File: src/applications/lgy/coe/form/config/chapters/loans/loanIntent.js (hunk 1 above)
Comment (Mottie) [reply src/applications/lgy/coe/form/config/chapters/loans/loanIntent.js]: Thanks!

**Pull request merged**